09-workload.md

```diff
@@ -20,6 +20,49 @@ The cluster now has an [Traefik configured with a TLS certificate](./08-secret-m
    kubectl wait -n a0008 --for=condition=ready pod --selector=app.kubernetes.io/name=aspnetapp --timeout=90s
    ```
 
+1. Deploy the Ingress resource
+
+   ```bash
+   cat <<EOF | kubectl create -f -
+   apiVersion: networking.k8s.io/v1
+   kind: Ingress
+   metadata:
+     name: aspnetapp-ingress
+     namespace: a0008
+     annotations:
+       kubernetes.io/ingress.allow-http: "false"
+       # defines controller implementing this ingress resource: https://docs.microsoft.com/en-us/azure/dev-spaces/how-to/ingress-https-traefik
+       # ingress.class annotation is being deprecated in Kubernetes 1.18: https://kubernetes.io/docs/concepts/services-networking/ingress/#deprecated-annotation
+       # For backwards compatibility, when this annotation is set, precedence is given over the new field ingressClassName under spec.
+       kubernetes.io/ingress.class: traefik-internal
+       traefik.ingress.kubernetes.io/router.entrypoints: websecure
+       traefik.ingress.kubernetes.io/router.tls: "true"
+       traefik.ingress.kubernetes.io/router.tls.options: default
+       traefik.ingress.kubernetes.io/router.middlewares: app-gateway-snet@file, gzip-compress@file
+   spec:
+     # ingressClassName: "traefik-internal"
+     tls:
+     - hosts:
+         - bu0001a0008-00.aks-ingress.$DOMAIN_NAME
+           # it is possible to opt for certificate management strategy with dedicated
+           # certificates for each TLS SNI route.
+           # In this Rereference Implementation for the sake of simplicity we use a
+           # wildcard default certificate added at Ingress Controller configuration level which is *.example.com
+           # secretName: <bu0001a0008-00-example-com-tls-secret>
+     rules:
+     - host: bu0001a0008-00.aks-ingress.$DOMAIN_NAME
+       http:
+         paths:
+         - path: /
+           pathType: Prefix
+           backend:
+             service:
+               name: aspnetapp-service
+               port:
+                 number: 80
+   EOF
+   ```
+
```

Comment:
We try to minimize inline yaml creation like this, (I know we have it in another spot here as well -- but trying to prevent more).  Any thoughts on maybe adding a Kustomization to the workload directory with a strategicMerge patch, then instead of this, we could do a quick `sed` command to replace a placeholder in that file, and instead of applying the workload with `-f` we could just `-k` that directory. There is a lot of "comment noise" in this file, which I think is best left in a file vs echo'd directly into a kubectl apply and to be lost.  Thoughts?

Reply:
I like that approach. I'll work on making the changes.

Reply:
Good start on this, I'm going to resolve this comment here, because my follow-on review has a slight tweak to the implementation, but the end result is the same.